src/collections.jl

```diff
@@ -8,7 +8,7 @@
 A collection of `items` (points or geometries) seen as a single domain.
 """
 struct Collection{Dim,T,I<:PointOrGeometry{Dim,T}} <: Domain{Dim,T}
-  items::Vector{I}
+    items::Union{Nothing,Vector{I}}
```

Comment:
Can't we somehow retain the type information and avoid the Union with Nothing? After all, we just need to return a zero length elsewhere right?

Reply:
Can't we just write something like `Collection([])` instead to construct an empty vector of items and then adjust the outer constructor accordingly with type annotations to avoid infinite recursions?

I think we need to open a more general discussion on Zulip or as an issue here to figure out how we should safely handle empty Domain types, not just Collection.